examples_gbs/run_tutorial_similarity.py

```diff
@@ -0,0 +1,203 @@
+
+print(similarity.sample_to_event([0, 4, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0], 3))
+
+##############################################################################
+# Now that we have mastered orbits and events, how can we make a feature vector? It was shown in
+# :cite:`schuld2019quantum` that one way of making a feature vector of a graph is to measure the
+# probabilities of events. Specifically, for a :math:`k` photon event :math:`E_{k, n_{\max}}`
+# with maximum count per mode :math:`n_{\max}` and corresponding probability :math:`p_{k,
+# n_{\max}}:=p_{E_{k, n_{\max}}}(G)` with respect to a graph :math:`G`, a feature vector can be
+# written as
+#
+# .. math::
+#     f_{\mathbf{k}, n_{\max}} = (p_{k_{1}, n_{\max}}, p_{k_{2}, n_{\max}}, \ldots , p_{k_{K}, n_{\max}}),
+#
+# where :math:`\mathbf{k} := (k_{1}, k_{2}, \ldots , k_{K})` is a list of different total photon
+# numbers.
+#
+# Calculating a feature vector
+# ----------------------------
+#
+# We provide two methods for calculating a feature vector of GBS event probabilities in
+# Strawberry Fields:
+#
+# 1. Through sampling
+# 2. Using a Monte Carlo estimate of the probability
+#
+# For the first method, all one needs to do is generate some GBS samples from the graph of
+# interest and feed through:
+
+print(similarity.feature_vector_sampling(m0, event_photon_numbers=[2, 4, 6], max_count_per_mode=2))
+
+##############################################################################
+# For the second method, suppose we want to calculate the event probabilities exactly rather than
+# through sampling. To do this, we consider the event probability :math:`p_{k, n_{\max}}` as the
+# sum over all sample probabilities in the event. As discussed in the GBS `tutorial
+# <gaussian_boson_tutorial>`, each sample probability is determined by the hafnian of a relevant
+# sub-adjacency matrix. While this is tough to calculate, what makes calculating :math:`p_{k,
+# n_{\max}}` really tough is the number of samples the corresponding event contains! For example,
+# the 17-photon event :math:`E_{k=6, n_{\max}=2}` contains the following number of samples:
+
+print(similarity.event_cardinality(6, 2, 17))
+
+##############################################################################
+# To avoid calculating a large number of sample probabilities, an alternative for calculating the
+# event probability is to perform a Monte Carlo approximation. Here, samples within an event are
+# generated uniformly at random and their resultant probabilities are calculated. If :math:`N`
+# samples :math:`\{S_{1}, S_{2}, \ldots , S_{N}\}` are generated, then the event probability can
+# be approximated as
+#
+# .. math::
+#     p(E_{k, n_{\max}}) \approx \frac{1}{N}\sum_{i=1}^N p(S_i) |E_{k, n_{\max}}|,
+#
+# with :math:`|E_{k, n_{\max}}|` denoting the cardinality of the event.
+#
+# This method can be accessed using the :func:`prob_event_mc` function. The 4-photon event is

```

Comment:
Is there a link to this function?

Reply:
https://github.com/XanaduAI/strawberryfields/blob/aacbadd8461f2aaf72e27a5d67beb1336cfff427/strawberryfields/gbs/similarity.py#L333

I decided not to show it explicitly in the tutorial because it take a while to evaluate. We can look back at this though.